spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonDbSchedulerAutoConfiguration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import com.github.kagkarlsson.scheduler.Scheduler;
+import com.github.kagkarlsson.scheduler.task.Task;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.config.Configuration;
+import org.axonframework.config.ConfigurationScopeAwareProvider;
+import org.axonframework.deadline.DeadlineManager;
+import org.axonframework.deadline.dbscheduler.DbSchedulerDeadlineDetails;
+import org.axonframework.deadline.dbscheduler.DbSchedulerDeadlineManager;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.eventhandling.scheduling.dbscheduler.DbSchedulerEventData;
+import org.axonframework.eventhandling.scheduling.dbscheduler.DbSchedulerEventScheduler;
+import org.axonframework.messaging.ScopeAwareProvider;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.tracing.SpanFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * Auto configuration class for the deadline manager and event scheduler using db-scheduler.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+@AutoConfiguration
+@ConditionalOnBean(Scheduler.class)
+@AutoConfigureAfter(value = {AxonServerAutoConfiguration.class, AxonTracingAutoConfiguration.class},
+        name = {"com.github.kagkarlsson.scheduler.boot.autoconfigure.DbSchedulerAutoConfiguration"})
+public class AxonDbSchedulerAutoConfiguration {
+
+    @Bean
+    Task<DbSchedulerEventData> dbSchedulerEventDataTask() {
+        return DbSchedulerEventScheduler.task();
```

Comment:
It needs to be initialized before the `DbSchedulerEventScheduler` because the `Scheduler` that need to be created to create the `DbSchedulerEventScheduler` needs all the tasks in the constuctor.